File: 8_kyu/Holiday_VI_Shark_Pontoon.py
def shark(pontoon_distance: float, shark_distance: float, you_speed: float, shark_speed: float, dolphin: bool) -> str:
    shark_time = shark_distance / shark_speed
    if dolphin:
        shark_time *= 2
    swim_time = pontoon_distance / you_speed
    return "Alive!" if swim_time <= shark_time else "Shark Bait!"
    

File: 8_kyu/Holiday_VI_Shark_Pontoon.rs
fn shark(
    pontoon_distance: f64,
    shark_distance: f64,
    you_speed: f64,
    shark_speed: f64,
    dolphin: bool,
) -> String {
    let dolphin_effect: f64 = if dolphin { 0.5 } else { 1.0 };
    let alive = shark_distance * you_speed > pontoon_distance * shark_speed * dolphin_effect;
    let result = if alive { "Alive!" } else { "Shark Bait!" };
    result.to_string()
}
